stock_quant_history/models/stock_quant_history_snapshot.py

```diff
@@ -182,7 +184,9 @@ def _generate_stock_quant_history(self):
                     ),
                     precision_rounding=move_line.product_id.uom_id.rounding,
                 )
-
+                pickings |= move_line.picking_id
+        # Lock all related pickings
+        pickings.write({"is_locked": True})
```

Comment:
I'm wondering if odoo perform write on record where is_locked is already True other wise I would probably filter before write:

```suggestion
        pickings.filtered(lambda picking: not picking.is_locked).write({"is_locked": True})
```

Reply:
Value check is performed here in fields.py
```    def get_records_different_from(self, records, field, value):
        """ Return the subset of ``records`` that has not ``value`` for ``field``. """
        field_cache = self._get_field_cache(records, field)
        if field.translate:
            lang = records.env.lang or 'en_US'

            def get_value(id_):
                cache_value = field_cache[id_]
                return None if cache_value is None else cache_value[lang]
        else:
            get_value = field_cache.__getitem__

        ids = []
        for record_id in records._ids:
            try:
                val = get_value(record_id)
            except KeyError:
                ids.append(record_id)
            else:
                if field.type == "monetary":
                    value = field.convert_to_cache(value, records.browse(record_id))
                if val != value:
                    ids.append(record_id)
        return records.browse(ids)
```

So answer is yes, but record by records even in a write with multiple records.